contract/src/main/kotlin/io/provenance/scope/loan/utility/ServicingDataFunctions.kt

```diff
@@ -83,3 +87,32 @@ internal fun ContractEnforcementContext.appendLoanStates(
         servicingDataBuilder.addLoanState(state)
     }
 }
+
+internal fun ContractEnforcementContext.appendServicingDocuments(
+    servicingDataBuilder: ServicingData.Builder,
+    newDocuments: Collection<DocumentMetadata>,
+) {
+    val existingDocumentMetadata = mutableMapOf<String, DocumentMetadata>()
+    servicingDataBuilder.docMetaList.forEach { documentMetadata ->
+        documentMetadata.checksum.takeIf { it.isSet() }?.checksum?.let { checksum ->
+            existingDocumentMetadata[checksum] = documentMetadata
+        }
+    }
+    val incomingDocumentChecksums = mutableMapOf<String, Boolean>()
+    for (newDocument in newDocuments) {
+        documentValidation(newDocument)
+        newDocument.checksum.checksum?.let { newDocChecksum ->
+            if (incomingDocumentChecksums[newDocChecksum] == true) {
+                raiseError("Loan document with checksum $newDocChecksum already provided in input")
```

Comment:
nit: The wording of this error threw me off a bit. I think saying something about how the same document was provided twice (or more than once) would make it a bit more clear.

Reply:
Currently, if your input has 4 documents with the checksum "foo", the exception thrown would have the substring "Loan document with checksum foo already provided in input (3 occurrences)". [Violation wording implying the "same document" is currently only used when the input has a checksum which is the same as the checksum of a document already in the record](https://github.com/provenance-io/loan-package-contracts/blob/154052dc83ace44e88a7314d1b0cb6f4047a8df0/contract/src/main/kotlin/io/provenance/scope/loan/utility/LoanContractValidations.kt#L13-L37), because I reasoned that that was the case where it is likely that it's actually the "same" document. I figured that in this case where the duplicates are among the input, that we should say it's about the input. So I would suggest changing it to something that still references the "input" rather than the language of "same document" - something like `"Checksum foo is not unique (3 occurrences)"`.